components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -91,6 +106,70 @@ - (void)setSelectedItem:(UITabBarItem *)selectedItem {
   }
 
   _selectedItem = selectedItem;
+  UIView *itemView = self.itemViews[itemIndex];
+  if ([itemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+    MDCTabBarViewItemView *selectedItemView = (MDCTabBarViewItemView *)itemView;
+    selectedItemView.image = selectedItem.selectedImage ?: selectedItem.image;
+  }
+}
+
+#pragma mark - Key-Value Observing (KVO)
+
+- (void)addObserversToTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item addObserver:self
+           forKeyPath:kImageKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+    [item addObserver:self
+           forKeyPath:kSelectedImageKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+    [item addObserver:self
+           forKeyPath:kTitleKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+  }
+}
+
+- (void)removeObserversFromTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item removeObserver:self forKeyPath:kImageKeyPath context:kKVOContextMDCTabBarView];
+    [item removeObserver:self forKeyPath:kSelectedImageKeyPath context:kKVOContextMDCTabBarView];
+    [item removeObserver:self forKeyPath:kTitleKeyPath context:kKVOContextMDCTabBarView];
+  }
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (context == kKVOContextMDCTabBarView) {
+    NSInteger selectedItemNum = 0;
+    for (NSUInteger i = 0; i < self.items.count; i++) {
+      UITabBarItem *item = self.items[i];
+      if (object == item) {
+        selectedItemNum = i;
+        break;
+      }
+    }
```

Comment:
This looks like `indexOfObject:`; can we use that method instead? If not, then consider renaming `selectedItemNum` to `itemIndex` because it is an index.

Reply:
Done.

Reply:
Done.